src/controller/python/chip/ChipBluezMgr.py

```diff
@@ -1060,20 +1075,37 @@ def disconnect_bg_implementation(self, **kwargs):
             self.service.destroy()
             self.service = None
 
+    def get_peripheral_devIdInfo(self, peripheral):
+        if not peripheral.ServiceData:
+            return None
+        servDataDict = dict(peripheral.ServiceData)
```

Comment:
can you iterate over ServiceData directly without creating a separate dictionary only for the purpose of iteration?

Happens a few more times below. what type is ServiceData? is it a tuple array (in which case you can iterate normally) or something else?

Reply:
the service data is a dbus.Dictionary
I will go through the python dbus doc and then cleanup these code in this pr.